Research/NFL_PlayByPlay/Capstone 2/nfl_plays_model.py

```diff
@@ -0,0 +1,147 @@
+import pandas as pd
+import numpy as np
+import glob
+import time
+from sklearn.ensemble import BaggingClassifier, RandomForestClassifier, GradientBoostingClassifier, ExtraTreesClassifier
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.svm import SVC
+from sklearn.cross_validation import train_test_split
+import warnings
+from sklearn.externals import joblib
+
+def gather_and_process_data(path, test_season=None):
+    # TODO Error raised if passed path is invalid
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        files = glob.glob(path)
+        seasons = pd.read_csv(files[0])
+        for x in files[1:]:
+            data = pd.read_csv(x)
+            seasons = seasons.append(data)
+    
+    # Pull specific columns for the model here
+    # 'time' column has the specific 'XX:XX' game clock that can be used if converted to a usable number
+    cdata = seasons[['posteam','DefensiveTeam','qtr', 'down', 'ydstogo', 'TimeUnder', 'yrdline100', 'ScoreDiff', 'PlayType', 'Season']]
+
+
+    def match(playtype):
+        valid_play_types = [
+        'Field Goal',
+        'Pass',
+        'Run',
+        'QB Kneel',
+        'Punt'
+        ]
+        return playtype in valid_play_types
+
+    cdata = cdata[cdata.PlayType.map(match)]
```

Comment:
You can probably use [`isin`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.isin.html) here instead.